src/pfe/portal/modules/LoadRunner.js

```diff
@@ -487,91 +505,99 @@ module.exports = class LoadRunner {
-    this.socket.on('connect', () => {
-      log.info('Loadrunner has connected')
-      this.up = true;
+    this.socket.on('connect', data => {
+      //if (data.projectID === this.project.projectID) {
+        log.info('Loadrunner has connected')
+        this.up = true;
+      //}
     });
 
     /**
     * Socket event for disconnect
     * Logs the error and starts reconnect procedure
     */
-    this.socket.on('disconnect', () => {
-      log.info('Loadrunner has disconnected')
-      // If this.up is false we're already trying to reconnect
-      if (this.up) {
-        // socket.io-client will automatically reconnect and trigger the connect event.
-        this.up = false;
+    this.socket.on('disconnect', data => {
+      if (data.projectID === this.project.projectID) {
+        log.info('Loadrunner has disconnected')
+        // If this.up is false we're already trying to reconnect
+        if (this.up) {
+          // socket.io-client will automatically reconnect and trigger the connect event.
+          this.up = false;
+        }
       }
     });
 
     /**
     * Socket event for connection timeout
     * Logs the error and starts reconnect procedure
     */
-    this.socket.on('connect_timeout', () => {
-      log.info('Loadrunner has timed out')
-      // If this.up is false we're already trying to reconnect
-      if (this.up) {
-        // socket.io-client will automatically reconnect and trigger the connect event.
-        this.up = false;
-      }
+    this.socket.on('connect_timeout', data => {
+      if (data.projectID === this.project.projectID) {
+        log.info('Loadrunner has timed out')
+        // If this.up is false we're already trying to reconnect
+        if (this.up) {
+          // socket.io-client will automatically reconnect and trigger the connect event.
+          this.up = false;
+        }
+      }  
     });
 
     /**
     * Socket event for error
     * Logs the error and starts reconnect procedure
     */
-    this.socket.on('error', (err) => {
-      log.error('LoadRunner socket error');
-      // A socket error could occur while a load run is not in progress.
-      if (this.project) {
-        this.project.loadInProgress = false;
-      }
-      log.error(err);
-      // If this.up is false we're already trying to reconnect
-      if (this.up) {
-        // socket.io-client will automatically reconnect and trigger the connect event.
-        this.up = false;
+    this.socket.on('error', (err, data) => {
+      if (data.projectID === this.project.projectID) {
+        log.error('LoadRunner socket error');
+        // A socket error could occur while a load run is not in progress.
+        if (this.project) {
+          this.project.loadInProgress = false;
+        }
+        log.error(err);
+        // If this.up is false we're already trying to reconnect
+        if (this.up) {
+          // socket.io-client will automatically reconnect and trigger the connect event.
+          this.up = false;
```

Comment:
`up` is only set to false if the error is for this project.  However I don't think loadrunner.up is ever read anywhere. (I think it was a hangover from when the LoadRunner.js was created from FileWatcher.js which at the time also used a socket.)
It might be less confusing to remove this. (It doesn't affect reconnecting.)